app/_components/MapContainer.tsx

```diff
@@ -76,8 +79,31 @@ function MapInner() {
 
   const user = useContext(AuthContext);
   const importantPinContext = useContext(ImportantPinContext);
+  const searchParams = useSearchParams();
+  const id = searchParams.get("id");
```

Comment:
Is there a way to make this restful? It might not be as intuitive if we look at the link `/map/id={num}`.
How can we clearly state that we are sharing a poi id?

Reply:
@J-Ariola good point! what if we use `/map/poicard/id={num}` instead?